bin/autodeploy.sh

```diff
@@ -0,0 +1,416 @@
+  sed -i 's/useCertManager: false/useCertManager: true/g' values/nginx-ingress-services/values.yaml
+  sed -i 's/certmasterEmail:/certmasterEmail: backend+wiabautodeploy@wire.com/g' values/nginx-ingress-services/values.yaml
+  sed -i "s/example.com/$TARGET_SYSTEM/" values/nginx-ingress-services/values.yaml
+
+  d kubectl create namespace cert-manager-ns
+  d helm upgrade --install -n cert-manager-ns --set 'installCRDs=true' cert-manager charts/cert-manager
+
+  d kubectl uncordon "$INGRESSNODE"
+
+  d helm upgrade --install nginx-ingress-services charts/nginx-ingress-services -f values/nginx-ingress-services/values.yaml
+
+  d kubectl get certificate
+
+  cp values/sftd/prod-values.example.yaml values/sftd/values.yaml
+  sed -i "s/webapp.example.com/webapp.$TARGET_SYSTEM/" values/sftd/values.yaml
+  sed -i "s/sftd.example.com/sft.$TARGET_SYSTEM/" values/sftd/values.yaml
+  sed -i 's/name: letsencrypt-prod/name: letsencrypt-http01/' values/sftd/values.yaml
+  sed -i "s/replicaCount: 3/replicaCount: 1/" values/sftd/values.yaml
+  d kubectl label node kubenode1 wire.com/role=sftd
+  d helm upgrade --install sftd ./charts/sftd --set 'nodeSelector.wire\.com/role=sftd' --set 'node_annotations="{'wire\.com/external-ip': '"$HOST_IP"'}"' --values values/sftd/values.yaml
+
+  ZREST_SECRET=$(grep -A1 turn values/wire-server/secrets.yaml | grep secret | tr -d '"' | awk '{print $NF}')
+
+  cat >values/coturn/values.yaml<<EOF
+nodeSelector:
+  wire.com/role: coturn
+
+coturnTurnListenIP: '192.168.122.23'
+coturnTurnRelayIP: '192.168.122.23'
+coturnTurnExternalIP: '$HOST_IP'
+EOF
+
+  cat >values/coturn/secrets.yaml<<EOF
+secrets:
+  zrestSecrets:
+    - "$ZREST_SECRET"
+EOF
+
+  d kubectl label node kubenode3 wire.com/role=coturn
+  d kubectl annotate node kubenode3 wire.com/external-ip="$HOST_IP"
+  d helm upgrade --install coturn ./charts/coturn --values values/coturn/values.yaml --values values/coturn/secrets.yaml
+}
+
+EXISTING_INSTALL=$(ssh -p "$SSH_PORT" "$SSH_USER"@webapp."$TARGET_SYSTEM" "ls /home/$DEMO_USER/wire-server-deploy-static-*.tgz 2>/dev/null" || true)
+EXISTING_VMS=$(ssh -p "$SSH_PORT" "$SSH_USER"@webapp."$TARGET_SYSTEM" "virsh list --all --name")
+EXISTING_CONTAINERS=$(ssh -p "$SSH_PORT" "$SSH_USER"@webapp."$TARGET_SYSTEM" "docker ps -q --all")
+
+if [[ "$EXISTING_INSTALL" ]]; then
+  msg ""
+  msg "WARNING: existing wire-server-deploy installation found: $EXISTING_INSTALL"
+  DO_SYSTEM_CLEANUP=true
+fi
+if [[ "$EXISTING_VMS" ]]; then
+  msg ""
+  msg "WARNING: existing libvirt VMs found: $EXISTING_VMS"
+  DO_SYSTEM_CLEANUP=true
+fi
+if [[ "$EXISTING_CONTAINERS" ]]; then
+  msg ""
+  msg "WARNING: existing Docker containers found."
+  DO_SYSTEM_CLEANUP=true
+fi
+
+if [ "$DO_SYSTEM_CLEANUP" = false ]; then
+  msg ""
+  msg "INFO: Target system clean, no previous wire-server-deploy installation found."
+fi
+if [ "$DO_SYSTEM_CLEANUP" = true ] && [ "$FORCE_REDEPLOY" = 0 ]; then
```

Comment:
**thought:** is there a reason to avoid an `else` branch and instead duplicate the conditions in the following `if`s?

```suggestion
elif [ "$FORCE_REDEPLOY" = 0 ]; then
```

Reply:
The existing logic accounts for the case of a "clean" target system where no resource removal is necessary, in which we don't need to jump into the system_cleanup() function. Otherwise the output looks like this:

```
INFO: Target system clean, no previous wire-server-deploy installation found.

INFO: Cleaning up all VMs, docker resources and wire-server-deploy files on wiab-autodeploy.wire.link.
```

It's just cosmetics, but I's like to have consistent script output.